samples/csharp/MicrosoftGraphSDK/Submission.cs

```diff
@@ -157,5 +188,115 @@ public class Submission
+            {
+                OdataType = "#microsoft.graph.educationFeedbackResourceOutcome",
+                FeedbackResource = new EducationWordResource
+                {
+                    OdataType = "#microsoft.graph.educationWordResource",
+                    DisplayName = "Document1.docx"
+                }
+            };
+
+            return await client.Education
+                .Classes[classId]
+                .Assignments[assignmentId]
+                .Submissions[submissionId]
+                .Outcomes
+                .PostAsync(educationOutcome);
+        }
+
+        /// <summary>
+        /// Update the properties of an educationOutcome object
+        /// </summary>
+        /// <param name="client">Microsoft Graph service client</param>
+        /// <param name="classId">User class id</param>
+        /// <param name="assignmentId">Assignment id in the class</param>
+        /// <param name="submissionId">Student submission id</param>
+        /// <param name="outcomeId">Outcome id</param>
+        /// <param name="educationOutcome">Education outcome object</param>
+        /// <returns>EducationOutcome</returns>
+
+        public static async Task<EducationOutcome> PatchOutcomeAsync(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId,
+            string submissionId,
+            string outcomeId,
+            EducationPointsOutcome educationOutcome)
+        {
+            try
+            {
+                return await client.Education
+                    .Classes[classId]
+                    .Assignments[assignmentId]
+                    .Submissions[submissionId]
+                    .Outcomes[outcomeId]
+                    .PatchAsync(educationOutcome);
+            }
+            catch (Exception ex)
+            {
+                throw new GraphException($"PatchOutcomeAsync call: {ex.Message}", ex, classId, assignmentId, submissionId, educationOutcome);
+            }
+        }
+
+        /// <summary>
+        /// List all the outcomes associated with a submission
+        /// </summary>
+        /// <param name="client">Microsoft Graph service client</param>
+        /// <param name="classId">User class id</param>
+        /// <param name="assignmentId">Assignment id in the class</param>
+        /// <param name="submissionId">Student submission id</param>
+        /// <returns>EducationOutcomeCollectionResponse</returns>
+        public static async Task<EducationOutcomeCollectionResponse> GetSubmissionOutcomes(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId,
+            string submissionId)
+        {
+            return await client.Education
+                .Classes[classId]
+                .Assignments[assignmentId]
+                .Submissions[submissionId]
+                .Outcomes
+                .GetAsync();
```

Comment:
wrap this code with a try,catch block as you did for the **PatchOutcomeAsync**

Reply:
wrapped the code in try catch block